cesmcatalog/gen_CESM_catalog.py

```diff
@@ -0,0 +1,170 @@
+#!/usr/bin/env python
+"""
+  Goal: a script that gets pointed to a CESM case directory and generates
+        an intake-esm catalog of output in the short-term archive directory
+
+  Current state: a script that gets pointed to a CESM case directory and prints
+                 some environment variables that will be necessary for above
+
+  NOTE: this script needs to be run in the CESM postprocessing python environment,
+        which is still based on python 2.7. Follow instructions at
+
+        https://github.com/NCAR/CESM_postprocessing/wiki/cheyenne-and-DAV-quick-start-guide
+"""
+
+import os
+import sys
+import subprocess
+import logging
+import argparse
+import fnmatch
+
+import pandas as pd
+
+################################################################################
+
+def _parse_args():
+  """ Wrapper for argparse, returns dictionary of arguments """
+
+  parser = argparse.ArgumentParser(description="Generate intake-esm catalog for a CESM case",
+                                   formatter_class=argparse.ArgumentDefaultsHelpFormatter)
+
+  parser.add_argument('-c', '--case-root', action='store', dest='case_root', required=True,
+                      help='CESM case root to generate intake-esm catalog for')
+
+  return parser.parse_args()
+
+################################################################################
+
+def _find_data(case_root):
+  logger = logging.getLogger(__name__)
+
+  # 1. change directory to case_root (if it exists)
+  try:
+    os.chdir(case_root)
+  except:
+    # TODO: set up logger instead of print statements
+    logger.error('{} does not exist'.format(case_root))
+    sys.exit(1)
+
+  # 2. Collect info on where time slice output is
+  if not os.path.isfile('./xmlquery'):
+    # TODO: set up logger instead of print statements
+    logger.error('Can not find xmlquery in {}'.format(case_root))
+    sys.exit(1)
+
+  run_config = dict()
+  for var in ['GET_REFCASE', 'RUN_REFCASE']:
+    run_config[var] = subprocess.check_output('./xmlquery --value {}'.format(var), shell=True)
+  DOUT_S = subprocess.check_output('./xmlquery --value DOUT_S', shell=True)
+  if DOUT_S == 'TRUE':
+    DOUT_S_ROOT = subprocess.check_output('./xmlquery --value DOUT_S_ROOT', shell=True)
```

Comment:
To be safe, we may need some error handling here by wrapping these lines a try block. What do you think?

Reply:
Yeah, error handling would be great. I'll add it to the list :)

Reply:
Now that we're relying on the `CIME.Case` object instead of running these via `subprocess`, the failure mode is returning `None` rather than the expected variable value; that's worth checking for, but it'll be an `if` statement instead of a `try` / `except` block.